sycl/source/handler.cpp

```diff
@@ -82,6 +82,15 @@ void handler::setHandlerKernelBundle(
   MImpl->MKernelBundle = NewKernelBundleImpPtr;
 }
 
+void handler::setHandlerKernelBundle(kernel Kernel) {
+  // Kernel may not have an associated kernel bundle if it is created from a
+  // program. As such, apply getSyclObjImpl directly on the kernel, i.e. not
+  //  the other way around: getSyclObjImp(Kernel->get_kernel_bundle()).
```

Comment:
I think this is not directly related to your change, but this comment makes me think there is a problem someplace.  Why is there no kernel bundle for a kernel created from a program?  The SYCL spec defines `kernel::get_kernel_bundle()` for all kernels, even those created from backend interop.

Reply:
Arguably there is, but since program is being removed I don't think it is worth the worry.

Reply:
OK, I was wondering if the issue about having no kernel bundle is limited to just the deprecated `program` case, or if it also included other cases like constructing a kernel from backend interop.  If it is only the `program` case, I agree there is no need to fix it now.  We should instead wait until `program` is removed.

Reply:
This problem should be unique to kernel bundles in kernels created from `sycl::program`. For `sycl::make_kernel` we either use a supplied kernel bundle (part of the L0 input type) or an empty "interop" kernel bundle.